File: event_front/src/app/services/invite/invite.service.ts
import { isAccepeted } from './../../model/invite/InviteAccepted';
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { inject, Injectable } from '@angular/core';
import { AuthService } from '../auth/auth.service';
import type { InviteRequest } from '../../model/invite/InviteRequest';
import { InviteResponse } from '../../model/invite/InviteResponse';

@Injectable({
  providedIn: 'root'
})
export class InviteService {
  private httpClient = inject(HttpClient);

  private url = "http://localhost:8080/api/invite";

  authService = inject(AuthService);

  registerInvite(inviteRequest: InviteRequest){

    const token = this.authService.getToken()

    const headers = new HttpHeaders({
      'Authorization': `Bearer ${token}`,
    });

    return this.httpClient.post(this.url + "/register", inviteRequest, {headers}).subscribe();
  }

  getInviteByUser(){
    const token = this.authService.getToken()

    const headers = new HttpHeaders({
      'Authorization': `Bearer ${token}`,
    });

    return this.httpClient.get<InviteResponse[]>(this.url + "/list-invite", {headers});
  }

  acceptedInvite(isAccepeted : isAccepeted){
    const token = this.authService.getToken()

    const headers = new HttpHeaders({
      'Authorization': `Bearer ${token}`,
    });

    return this.httpClient.post(this.url + "/isAccepted", isAccepeted,  {headers}).subscribe();
  }
}

File: event_back/src/main/java/br/thullyoo/event_back/service/InviteService.java
package br.thullyoo.event_back.service;

import br.thullyoo.event_back.dto.request.invite.InviteRequest;
import br.thullyoo.event_back.dto.request.invite.IsAcceptedInvite;
import br.thullyoo.event_back.dto.response.invite.InviteResponse;
import br.thullyoo.event_back.model.Invite;

import java.util.List;
import java.util.UUID;

public interface InviteService {

    public Invite registerInvite(InviteRequest inviteRequest);

    public List<InviteResponse> getInviteByUser();

    public void isAcceptedInvite(IsAcceptedInvite isAcceptedInvite);
}
